File: httplib/src/main/java/com/dolphkon/httplib/base/BaseMvpActivity.java
package com.dolphkon.httplib.base;
import android.os.Bundle;
import androidx.annotation.Nullable;

/**
 * ****************************************************
 * Project: WanandroidClient
 * PackageName: com.dolphkon.httplib.base
 * ClassName：BaseMvpActivity
 * Author: kongdexi
 * Date: 2020/10/12 20:23
 * Description:TODO
 * *****************************************************
 */
public abstract class BaseMvpActivity<P> extends BaseActivity implements BaseView{
    public P mPresenter;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        if (mPresenter == null) {
            mPresenter = createPresenter();
        }

        if (mPresenter == null) {
            throw new NullPointerException("mPresenter 不能为空!");
        }
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onDestroy() {
        mPresenter=null;
        super.onDestroy();
    }

    protected abstract P createPresenter();

    @Override
    public void showToast(String msg) {
        toast(msg);
    }

    @Override
    public void showToast(int msgId) {
        toast(msgId);
    }

    @Override
    public void showLoading() {
        showProgressDialog();
    }

    @Override
    public void showLoading(String msg) {
        showProgressDialog(msg);
    }

    @Override
    public void showLoading(int msgid) {
        showProgressDialog(msgid);
    }

    @Override
    public void hideLoading() {
        dismissProgressDialog();
    }
}

File: httplib/src/main/java/com/dolphkon/httplib/base/BaseMvpActivity.kt
package com.dolphkon.httplib.base

import android.os.Bundle

/**
 * ****************************************************
 * Project: WanandroidClient
 * PackageName: com.dolphkon.httplib.base
 * ClassName：BaseMvpActivity
 * Author: kongdexi
 * Date: 2020/10/12 20:23
 * Description:TODO
 * *****************************************************
 */
abstract class BaseMvpActivity<P : BasePresenter> : BaseActivity(), BaseView {
    @JvmField
    var mPresenter: P? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        if (mPresenter == null) {
            mPresenter = createPresenter()
        }
        if (mPresenter == null) {
            throw NullPointerException("mPresenter 不能为空!")
        }
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        mPresenter?.unDisposable()
        mPresenter = null
        super.onDestroy()
    }

    protected abstract fun createPresenter(): P

    override fun showLoading() {
        showProgressDialog()
    }

    override fun showLoading(msg: String?) {
        showProgressDialog(msg)
    }

    override fun showLoading(msgid: Int) {
        showProgressDialog(msgid)
    }

    override fun hideLoading() {
        dismissProgressDialog()
    }
}